Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -1167,10 +1186,75 @@ WarpXOpenPMDPlot::GetMeshCompNames (int meshLevel,
     }
 
     if ( 0 == meshLevel )
-      return;
+        return;
 
     field_name += std::string("_lvl").append(std::to_string(meshLevel));
 }
+/* Find length of field name in varname and get the theta mode index, if varname = fieldName_mode_realOrImag
+ *
+ * @param varname [IN]: name of the field variable being parsed
+ * @returns varname fieldName if varname = fieldName_mode_realOrImag, otherwise, varname and
+ * if varname = fieldName_modeNumber_realOrImag, returns 2 * mode - 1 + (realOrImag == 'imag'), otherwise, -1
+ *
+ * Examples :
+ * rho -> 3, -1
+ * rho_0_real -> 3, 1
+ * Er_1_real -> 2, 2
+ * divB_1_imag -> 4, 3
+ */
+std::tuple<std::string, int>
+GetFieldNameModeInt (const std::string& varname)
+{
+    // mode_index = -1 if varname isn't of form fieldName_mode_realOrImag
+    // mode_index = 2 * mode - 1 + (realOrImag == 'imag')
+    // in either case, there is a -1 in mode_index
+    int mode_index = -1;
+
+    std::regex e_real_imag ("(.*)_([0-9]*)_(real|imag)");
+    std::smatch sm;
+    std::regex_match (varname, sm, e_real_imag, std::regex_constants::match_default);
+
+    if (sm.size() != 4 ) {
+        return std::make_tuple(varname, mode_index);
+    } else {
+        // sm = [varname, field_name, mode, real_imag]
+        int mode = std::stoi(sm[2]);
+        if (mode == 0) {
+            mode_index = 0;
+        } else {
+            if (sm[3] == "imag") {
+                mode_index += 1;
+            }
+            mode_index += 2 * mode;
+        }
+
+        return std::make_tuple(std::string(sm[1]), mode_index);
+    }
+}
+/* Transpose diagnostic data from WarpX/AMReX/Fortran order to openPMD (RZ) / C order
+ * NOTE : duplicates chunk in memory and is NOT optimized
+ *
+ * @param varname [IN]: pointer to the data to be transposed
+ * @param local_data [OUT]: pointer to the place to store transpoed copy of the chunk
+ * @param local_box [OUT]: reference to chunk `amrex::Box` for chunk_size information
```

Comment:
The Doxygen `@param` syntax is like this: https://www.doxygen.nl/manual/commands.html#cmdparam
```suggestion
 * @param[in] varname pointer to the data to be transposed
 * @param[out] local_data pointer to the place to store transpoed copy of the chunk
 * @param[out] local_box reference to chunk `amrex::Box` for chunk_size information
```